### ISSUE TITLE
Trailing space on selector breaks parser

### ISSUE THREAD

**Issue by AmazingDreams** [jhy/jsoup#1082]
I have a weird issue that occurs in 1.11.1+ but not in 1.10.3. The document selector string is cleared somehow, but I cannot find where and how.

```scala
if (selector == null || selector.isEmpty) {
  return
}

Logger.debug(s"selector: $selector")

val selectedElements = doc.select(selector) // Exception
```

```
[debug] application - selector: .classA 
[info] org.jsoup.select.Selector$SelectorParseException: Could not parse query '': unexpected token at ''
[info]   at org.jsoup.select.QueryParser.findElements(QueryParser.java:204)
[info]   at org.jsoup.select.QueryParser.parse(QueryParser.java:59)
[info]   at org.jsoup.select.QueryParser.parse(QueryParser.java:42)
[info]   at org.jsoup.select.QueryParser.combinator(QueryParser.java:87)
[info]   at org.jsoup.select.QueryParser.parse(QueryParser.java:69)
[info]   at org.jsoup.select.QueryParser.parse(QueryParser.java:42)
[info]   at org.jsoup.select.Selector.select(Selector.java:90)
[info]   at org.jsoup.nodes.Element.select(Element.java:363)
[info]   at utils.html.CssInliner$.applyStylesToElements(CSSInliner.scala:68)
[info]   at utils.html.CssInliner$.$anonfun$inlineStyles$1(CSSInliner.scala:34)
```

**Comment by AmazingDreams** [comment]
While making this issue I realized there was a trailing space on the selector, which broke the parser. Using `trim()` to clear the trailing space fixed the issue.

I fixed my specific problem but I can imagine that this is not supposed to happen. See what you do with this issue.

**Comment by jhy** [comment]
Thanks. This is a dupe of (fixed) #1274
